test_requests.py

```diff
@@ -188,6 +188,28 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert '/relative-redirect/20' in e.request.url
+            assert e.request.url == e.response.url
```

Comment:
I think it could help catch off by one logic where the previous redirect request is returned instead of the one used to get the last request.  I'll swap it for a more direct comparison instead.